src/wasm-type.h

```diff
@@ -177,6 +176,47 @@ class Type {
   }
 
   std::string toString() const;
+
+  Type& operator[](size_t i) {
```

Comment:
We have to make sure that Types remain immutable. This change may require additional consts down in the iterator implementation as well.

```suggestion
  const Type& operator[](size_t i) const {
```

Reply:
Worked as-is :)